models/NASFPN/builder.py

```diff
@@ -0,0 +1,522 @@
+import math
+import mxnet as mx
+import mxnext as X
+
+from mxnext.complicate import normalizer_factory
+from symbol.builder import Backbone, Neck
+from models.retinanet.builder import RetinaNetHead
+from models.retinanet.builder import RetinaNetNeck
+
+
+def merge_sum(f1, f2, name):
+    """
+    :param f1: feature 1
+    :param f2: feature 2
+    :param name: name
+    :return: sum(f1, f2)
+    """
+    return mx.sym.ElementWiseSum(f1, f2, name=name + '_sum')
+
+def merge_gp(f1, f2, name):
+    """
+    :param f1: feature 1, attention feature
+    :param f2: feature 2, major feature
+    :param name: name
+    :return: global pooling fusion of f1 and f2
+    """
+    gp = mx.sym.Pooling(f1, name=name + '_gp', kernel=(1, 1), pool_type="max", global_pool=True)
+    gp = mx.sym.Activation(gp, act_type='sigmoid', name=name + '_sigmoid')
+    fuse_mul = mx.sym.broadcast_mul(f2, gp, name=name + '_mul')
+    fuse_sum = mx.sym.ElementWiseSum(f1, fuse_mul, name=name + '_sum')
+    return fuse_sum
+
+
+def reluconvbn(data, num_filter, init, norm, name, prefix):
+    """
+    :param data: data
+    :param num_filter: number of convolution filter
+    :param init: init method of conv weight
+    :param norm: normalizer
+    :param name: name
+    :return: relu-3x3conv-bn
+    """
+    data = mx.sym.Activation(data, name=name+'_relu', act_type='relu')
+    weight = mx.sym.var(name=prefix + name + "_weight", init=init)
+    bias = mx.sym.var(name=prefix + name + "_bias")
+    data = mx.sym.Convolution(data, name=prefix + name, weight=weight, bias=bias, num_filter=num_filter, kernel=(3, 3), pad=(1, 1), stride=(1, 1))
+    data = norm(data, name=name+'_bn')
+    return data
+
+
+class NASFPNNeck(Neck):
+    def __init__(self, pNeck):
+        super(NASFPNNeck, self).__init__(pNeck)
+        self.p = pNeck
```

Comment:
DO NOT do this. This will void the patch_config. Just let the base class handle the `self.p` assginment.